clang/test/CXX/drs/dr18xx.cpp

```diff
@@ -25,7 +25,7 @@ template <int &> struct S {}; // #dr1801-S
 S<i> V; // #dr1801-S-i
 // cxx98-14-error@-1 {{non-type template argument does not refer to any declaration}}
 //   cxx98-14-note@#dr1801-S {{template parameter is declared here}}
-// since-cxx17-error@#dr1801-S-i {{non-type template argument refers to subobject '.i'}}
+// cxx17-error@#dr1801-S-i {{non-type template argument refers to subobject '.i'}}
```

Comment:
I'm not sure about this. [Here](https://www.open-std.org/jtc1/sc22/wg21/docs/cwg_defects.html#1801) it is stated that this should be ill-formed, with a reference to [N4268](https://open-std.org/JTC1/SC22/WG21/docs/papers/2014/n4268.html). But it seems that it is not actual after P1907R1.

Reply:
Quoting P1907R1 wording changes:
> For a non-type template-parameter of reference or pointer type <...> the reference or pointer value shall not refer to or be the address of (respectively):
> ~a subobject (6.7.2),~
> <...>
> a subobject (6.7.2) of one of the above.

My reading is that you indeed can reference subobjects in NTTP since 20, unless they fall into one of the narrow categories listed in that paragraph. The change you made is correct, unless P1907R1 was adopted as a DR.

Sorry I forgot to check current wording when I added the test.